packages/ibc-utils/src/generic.rs

```diff
@@ -0,0 +1,43 @@
+use std::collections::BTreeMap;
+
+use cosmwasm_std::{coin, CosmosMsg, Env, IbcTimeout, IbcTimeoutBlock, StdResult, Timestamp};
+
+use crate::types::PacketForwardMiddlewareConfig;
+
+// Default timeout for IbcTransfer is 600 seconds
+const DEFAULT_TIMEOUT_TIMESTAMP: u64 = 600;
+
+#[allow(clippy::too_many_arguments)]
+pub fn ibc_send_message(
+    env: Env,
+    channel: String,
+    to: String,
+    denom: String,
+    amount: u128,
+    memo: String,
+    timeout_height: Option<u64>,
+    timeout_timestamp: Option<u64>,
+    _denom_to_pfm_map: BTreeMap<String, PacketForwardMiddlewareConfig>,
+) -> StdResult<CosmosMsg> {
+    Ok(CosmosMsg::Ibc(cosmwasm_std::IbcMsg::Transfer {
+        channel_id: channel,
+        to_address: to,
+        amount: coin(amount, denom),
+        timeout: match (timeout_height, timeout_timestamp) {
+            (Some(height), None) => IbcTimeout::with_block(IbcTimeoutBlock {
+                revision: 2,
+                height,
+            }),
+            (None, Some(timestamp)) => IbcTimeout::with_timestamp(Timestamp::from_nanos(timestamp)),
+            (Some(height), Some(timestamp)) => IbcTimeout::with_both(
+                IbcTimeoutBlock {
+                    revision: 2,
+                    height,
+                },
+                Timestamp::from_nanos(timestamp),
+            ),
+            _ => IbcTimeout::with_timestamp(env.block.time.plus_seconds(DEFAULT_TIMEOUT_TIMESTAMP)),
+        },
+        memo: Some(memo),
```

Comment:
quick question? cant we do the denom_to_pfm here?

Reply:
The doc comment for that field says this:

> This field is only supported on chains with CosmWasm >= 2.0 and silently ignored on older chains.
> If you need support for both 1.x and 2.x chain with the same codebase, it is recommended to use `CosmosMsg::Stargate` with a custom MsgTransfer protobuf encoder instead.

Reply:
I see, makes sense